src/fides/api/task/create_request_tasks.py

```diff
@@ -429,41 +429,59 @@ def run_access_request(
-            erasure_end_nodes: List[CollectionAddress] = list(graph.nodes.keys())
-            persist_initial_erasure_request_tasks(
-                session, privacy_request, traversal_nodes, erasure_end_nodes, graph
+        try:
+            logger.info("Building access graph for {}", privacy_request.id)
+            traversal: Traversal = Traversal(graph, identity)
+
+            # Traversal.traverse populates traversal_nodes in place, adding parents and children to each traversal_node.
+            traversal_nodes: Dict[CollectionAddress, TraversalNode] = {}
+            end_nodes: List[CollectionAddress] = traversal.traverse(
+                traversal_nodes, collect_tasks_fn
+            )
+            # Save Access Request Tasks to the database
+            ready_tasks = persist_new_access_request_tasks(
+                session, privacy_request, traversal, traversal_nodes, end_nodes, graph
             )
 
-        # cache a map of collections -> data uses for the output package of access requests
-        privacy_request.cache_data_use_map(
-            format_data_use_map_for_caching(
-                {
-                    coll_address: tn.node.dataset.connection_key
-                    for (coll_address, tn) in traversal_nodes.items()
-                },
-                connection_configs,
+            if (
+                policy.get_rules_for_action(action_type=ActionType.erasure)
+                and not privacy_request.erasure_tasks.count()
+            ):
+                # If applicable, go ahead and save Erasure Request Tasks to the Database.
+                # These erasure tasks aren't ready to run until the access graph is completed
+                # in full, but this makes sure the nodes in the graphs match.
+                erasure_end_nodes: List[CollectionAddress] = list(graph.nodes.keys())
+                persist_initial_erasure_request_tasks(
+                    session, privacy_request, traversal_nodes, erasure_end_nodes, graph
+                )
+
+            # cache a map of collections -> data uses for the output package of access requests
+            privacy_request.cache_data_use_map(
+                format_data_use_map_for_caching(
+                    {
+                        coll_address: tn.node.dataset.connection_key
+                        for (coll_address, tn) in traversal_nodes.items()
+                    },
+                    connection_configs,
+                )
             )
-        )
+        except TraversalError as err:
+            logger.error(
+                "TraversalError encountered for privacy request {}. Error: {}",
+                privacy_request.id,
+                err,
+            )
+            errors = ", ".join(err.errors)
+            privacy_request.add_error_execution_log(
+                session,
+                connection_key=errors,
+                dataset_name=errors,
+                collection_name=errors,
```

Comment:
Your dataset name and collection name are incorrect here, same for the other example. I understand that this error is being thrown outside of a typical node, and there could be many dataset names and collection names, but I don't think we should try to cram everything in here.

In this case, instead of creating an "execution log", I might suggest an "audit log" which we've used in the past for errors on privacy requests that pertain to multiple nodes.



```
fides=# select dataset_name, collection_name from executionlog;
       dataset_name        |      collection_name      
---------------------------+---------------------------
 prod-2644-example:address | prod-2644-example:address
 prod-2644-example:address | prod-2644-example:address

```

Reply:
ah I hadn't seen these audit logs , I'll look into that! 😄

Reply:
i'm not sure if it's exactly what you need my concern is that a list of strings under dataset_name or collection _name wasn't the intent of this column, but if audit log is also not a good fit, we can pick the closest one

Reply:
yeah I can take a look. I also thought of adding one ExecutionLog per failing node, but was afraid that it would crowd the UI a bit. Do you think that'd be a better approach?

Reply:
That's a good idea as well, I think that would generally work unless that had hundreds+ unconnected nodes. I guess my preference might be a 1) single audit log 2) execution log per failing node 3) single execution log with just the first unconnected node 3) single execution log with all the unconnected nodes.